src/validations/html/rules/MarkupOutsideContainer.js

```diff
@@ -0,0 +1,75 @@
+// This rule checks for any unexpected tags or text that
+// don't get caught by other validators, specifically:
+//
+// * Any tags or text outside the <html> element
+// * Any text that is inside the <html> element but not part
+//   of the <head> or <body> tags
+
+import findLastIndex from 'lodash-es/findLastIndex';
+
+import Code from './Code';
+
+export default class MismatchedTag {
+  constructor() {
+    this._openTagStack = [];
+    this._locationsWithOutsideTags = [];
+  }
+
+  openTag(location, {name}) {
+    if (
+      this._isOutsideAcceptableContainer() &&
```

Comment:
Let’s decompose this into something like `_isOutsideDocumentElement()` and `isDirectChildOfDocumentElement()` and then pair each directly with the below acceptable tag names

Reply:
Done